pkg/ucp/integrationtests/testserver/testserver.go

```diff
@@ -279,10 +282,24 @@ func StartWithETCD(t *testing.T, configureModules func(options modules.Options)
 	queueClient, err := queueProvider.GetClient(ctx)
 	require.NoError(t, err)
 
+	router := chi.NewRouter()
+	router.Use(servicecontext.ARMRequestCtx(pathBase, "global"))
+
+	app := middleware.NormalizePath(router)
+	server := httptest.NewUnstartedServer(app)
+	server.Config.BaseContext = func(l net.Listener) context.Context {
+		return ctx
+	}
+
+	// Unfortunately the server doesn't populate 'server.URL' until it is started, so we have to build it ourselves.
+	address := "http://" + server.Listener.Addr().String()
+	connection, err := sdk.NewDirectConnection(address + pathBase)
+	require.NoError(t, err)
+
```

Comment:
Needed to reorder some things here to get the URL of the server. The backend controllers call back into UCP to initiate a cascading delete, so the URL is required.